src/dvc_data/hashfile/build.py

```diff
@@ -116,6 +116,9 @@ def _build_tree(
         # NOTE: relpath might not exist
         relpath = path
 
+    if path.endswith(fs.sep):
+        path = path.rstrip(fs.sep)
```

Comment:
no harm in just doing
```suggestion
    path = path.rstrip(fs.sep)
```

Reply:
also, might want do do it above, before relpath